src/models/signature.rs

```diff
@@ -23,28 +24,20 @@ impl TryFrom<StarknetSignature> for EthSignature {
     type Error = StarknetSignatureError;
 
     fn try_from(value: StarknetSignature) -> Result<Self, Self::Error> {
-        let r_low: U256 = Felt252Wrapper::from(
-            *value.0.get(0).ok_or_else(|| StarknetSignatureError::MissingSignatureParamsError("r".to_string()))?,
-        )
-        .into();
-        let r_high: U256 = Felt252Wrapper::from(
-            *value.0.get(1).ok_or_else(|| StarknetSignatureError::MissingSignatureParamsError("r".to_string()))?,
-        )
-        .into();
+        macro_rules! signature_param {
```

Comment:
it feels a bit weird to declare a macro inside a function, don't you think?
is it idiomatic rust?

Reply:
I think it's ok, the idea here is that we don't use it anywhere else then inside this function, so it doesn't really make sense to have it available anywhere else.